src/esi/Expression.cc

```diff
@@ -215,6 +218,7 @@ evalnegate(stackmember * stack, int *depth, int whereAmI, stackmember * candidat
     /* copy down */
     --(*depth);
 
+    Must2(*depth < STACK_DEPTH_LIMIT-1, "expression too complex");
```

Comment:
This Must() looks odd because we neither _increase_ `depth` nor add stack elements here. We are actually removing one stacked element AFAICT. Please double check.

Reply:
Good catch. Thanks. This was supposed to handle the underflow cases.